File: scraper.js
const puppeteer = require("puppeteer");

const run = async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();

  await page.goto("https://mhst.kiranico.com/mhs2/data/monsties");

  // Table that has all the other tables with the desired data
  const parentTable = await page.$x(
    "/html/body/div/div/div[2]/div[2]/div[1]/div/table/tbody"
  );

  console.log("Searching table 1");
  const tables = await parentTable[0].$$eval("tr", (rows) => {
    /**
     *
     * @param {Element} row
     * @returns monster name
     */
    // const getMonsterName = (row) => row.children[0].textContent;
    console.log("iterating over rows");
    // rows.forEach((row) => {
    //   // console.log(row.children[0].textContent);
    // });
  });
};

run();

File: scraper.py
import requests
from bs4 import BeautifulSoup

url = "https://mhst.kiranico.com/mhs2/data/monsties"
req = requests.get(url)
soup = BeautifulSoup(req.content, 'html.parser')


def get_name(row):
    return row.div.get_text()


for row in list(soup.tbody.children)[:1]:
    for child in row.td.children:
        print(child.get_text("\n", strip=True))
